### PR TITLE
RN-359: Adds edited indicator to post.

### PR DESCRIPTION
**Pull request by mkraft**
#### Summary
Adds edited indicator to posts inline with messages. 

#### Ticket Link
[RN-359](https://mattermost.atlassian.net/browse/RN-359)

#### Checklist
- [ ] Added or updated unit tests
- [x] Has UI changes

#### Device Information
* iPhone 6 (iOS 11.1) emulator
* Pixel (Android 8.0.0) emulator

#### Screenshots
simulator screen shot - iphone 6 - 2017-11-17 at 15 51 09
simulator screen shot - iphone 6 - 2017-11-17 at 15 51 05
screenshot_1510951903
screenshot_1510951894

#### Awaiting PR
Note that there is a [PR pending](https://github.com/hmhealey/commonmark-react-renderer/pull/2) dependency. I will update the `package.json` when that PR has been merged.

### CODE REVIEW

**Comment by mkraft** [comment]
Here's the webapp for reference. 

I just noticed the edited indicator text colour varies slightly in this branch, so I'm going to fix that.

screen shot 2017-11-17 at 2 08 51 pm

**Comment by mkraft** [comment]
Updated opacity.

**Comment by mkraft** [comment]
PR into external package merged; no longer awaiting PR.

File: app/components/markdown/index.js [hunk 1]
```diff
@@ -303,9 +308,29 @@ export default class Markdown extends PureComponent {
...
 
+    renderEditedIndicator = ({context}) => {
+        let message = '';
+        let opacity = Platform.select({ios: 0.7, android: 1});
+        if (context[0] === 'paragraph') {
+            message = ' ';
+            opacity = Platform.select({ios: 0.5, android: 0.6});
+        }
+        message += this.props.intl.formatMessage({id: 'post_message_view.edited', defaultMessage: '(edited)'});
+        return <Text style={[style.editedIndicatorText, {opacity}]}>{message}</Text>;
```
Comment (enahum) [inline app/components/markdown/index.js]: Here you can use the `FormattedText` component that will do the localization for you, and then you can get rid of the intl prop

File: app/components/post_body/post_body.js [hunk 2]
```diff
@@ -224,13 +226,16 @@ class PostBody extends PureComponent {
                     <View style={[{flex: 1}, (isPendingOrFailedPost && style.pendingPost)]}>
                         <Markdown
                             baseTextStyle={messageStyle}
-                            textStyles={textStyles}
                             blockStyles={blockStyles}
+                            intl={intl}
```
Comment (enahum) [inline app/components/post_body/post_body.js]: you won't be needing this

Comment (mkraft) [reply app/components/post_body/post_body.js]: @enahum Making the change.

**Comment by mkraft** [comment]
@enahum Let me see if it can be vertically aligned with the emoji.

**Comment by mkraft** [comment]
@enahum Don't merge yet my change broke the Android styling.

**Comment by mkraft** [comment]
@asaadmahmood Here are the screenshots of Android and iOS. You can see that the misalignment of text beside an emoji is an existing production bug because it occurs without any edited indicator so it's out of scope of this story. I'll open a bug to fix it for Android.

simulator screen shot - iphone 6 - 2017-11-20 at 12 04 52
screenshot_1511197474

**Comment by mkraft** [comment]
@asaadmahmood Opened [RN-525](https://mattermost.atlassian.net/browse/RN-525).

**Comment by hmhealey** [comment]
Is the vertical alignment of the emoji with the edited indicator different than the alignment with regular text? I can tell you from experience that changing the alignment of those is a nightmare. iOS at least lets you change it, but I don't think you can with Android

**Comment by hmhealey** [comment]
Oh, you added another comment since I opened this page. We can give the vertical alignment another shot if they've added anything better, but I can almost guarantee that it's as good as it can get right now

File: app/components/markdown/index.js [hunk 3]
```diff
@@ -303,9 +307,38 @@ export default class Markdown extends PureComponent {
...
+        let opacity = Platform.select({ios: 0.7, android: 1});
+        if (context[0] === 'paragraph') {
+            spacer = ' ';
+            opacity = Platform.select({ios: 0.5, android: 0.6});
+        }
+        const styles = [style.editedIndicatorText, {opacity}];
+        if (Platform.OS === 'ios') {
+            styles.push(this.computeTextStyle(this.props.baseTextStyle, context));
+        }
+        return (<Text style={styles}>
```
Comment (hmhealey) [inline app/components/markdown/index.js]: Small style thing, we usually have the brackets on a separate line when the component spans multiple lines like
```
return (
    <Text
     />
);
```

File: app/components/markdown/index.js [hunk 4]
```diff
@@ -303,9 +307,38 @@ export default class Markdown extends PureComponent {
...
+            <Text>{spacer}</Text>
```
Comment (hmhealey) [inline app/components/markdown/index.js]: The extra `<Text>` around `{spacer}` may not be necessary

Comment (enahum) [reply app/components/markdown/index.js]: it is, if we don't add it in a Text component RN will complain, I'm almost sure

**Pull request merged**